File: backend/app/schemas/knowledge.py
"""Knowledge base API schemas."""

from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeUploadResponse(BaseModel):
    doc_id: str
    doc_name: str
    status: str
    upload_time: datetime
    chunk_count: int = 0


class KnowledgeDocumentSummary(BaseModel):
    doc_id: str
    doc_name: str
    description: str | None = None
    filename: str
    status: str
    chunk_count: int
    upload_time: datetime


class KnowledgeDocumentListResponse(BaseModel):
    items: list[KnowledgeDocumentSummary] = Field(default_factory=list)
    total: int = 0


class KnowledgeSearchHit(BaseModel):
    doc_id: str
    doc_name: str | None = None
    chunk_text: str
    similarity_score: float
    chunk_index: int | None = None


class KnowledgeSearchResponse(BaseModel):
    query: str
    hits: list[KnowledgeSearchHit] = Field(default_factory=list)
    total: int = 0


class KnowledgeDeleteResponse(BaseModel):
    doc_id: str
    deleted: bool
    message: str

File: frontend/src/types/knowledge.ts
export type KnowledgeUploadResponse = {
  doc_id: string;
  doc_name: string;
  status: string;
  upload_time: string;
  chunk_count: number;
};

export type KnowledgeDocument = {
  doc_id: string;
  doc_name: string;
  description: string | null;
  filename: string;
  status: string;
  chunk_count: number;
  upload_time: string;
};

export type KnowledgeDocumentListResponse = {
  items: KnowledgeDocument[];
  total: number;
};

export type KnowledgeSearchHit = {
  doc_id: string;
  doc_name: string | null;
  chunk_text: string;
  similarity_score: number;
  chunk_index: number | null;
};
